packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -422,30 +422,29 @@ export class PeerPool extends EventEmitter {
 			peers: this._pickRandomDiscoveredPeers(MAX_PEER_LIST_BATCH_SIZE)
 				.map(
 					(peer: Peer): ProtocolPeerInfo | undefined => {
-						const peerDetailedInfo: P2PDiscoveredPeerInfo | undefined =
-							peer.detailedPeerInfo;
+						const peerDiscoveredInfo: P2PDiscoveredPeerInfo | undefined =
+							peer.peerInfo;
 
-						if (!peerDetailedInfo) {
+						if (!peerDiscoveredInfo) {
 							return undefined;
 						}
 
 						// The options property is not read by the current legacy protocol but it should be added anyway for future compatibility.
 						return {
-							broadhash: peerDetailedInfo.options
-								? (peerDetailedInfo.options.broadhash as string)
+							broadhash: peerDiscoveredInfo.options
```

Comment:
`peerDiscoveredInfo.options` looks like it's not supposed to be here.

Reply:
Yeah I missed this one